Formula/lotus.rb

```diff
@@ -9,14 +9,57 @@ class Lotus < Formula
   depends_on "pkg-config" => :build
   depends_on "go"         => :build
   depends_on "jq"         => :build
+  depends_on "bzr"        => :build
+  depends_on "rustup-init" => :build
   depends_on "hwloc"
 
+  def arch_M1
+    IO.popen(['arch']).read.strip === 'arm64'
+  end
+
+  def install_submodules
+    system "git clone --recurse-submodules https://github.com/filecoin-project/filecoin-ffi.git extern/filecoin-ffi"
+    system "git clone --recurse-submodules https://github.com/filecoin-project/serialization-vectors.git extern/serialization-vectors"
+    system "git clone --recurse-submodules https://github.com/filecoin-project/test-vectors.git extern/test-vectors"
+    system "touch build/.update-modules"
+  end
+
+  def build_filecoin_ffi
+    ## First install rust to prevent Error on filecoin-ffi build:
+    ##    ++ command -v rustup
+    ##    + '[' -x '' ']'
+    ##    + echo '[build_from_source] Error: rustup is not installed.'
+    ##    [build_from_source] Error: rustup is not installed.
+    ##    + echo '[build_from_source] install Rust toolchain installer to resolve this problem.'
+    ##    [build_from_source] install Rust toolchain installer to resolve this problem.
+    ##    + exit 1
+
+    # Set cargo_home to local directory
+    system 'mkdir -p extern/cargo_home'
+    ENV['CARGO_HOME'] = 'extern/cargo_home'
+    system 'rustup-init -y'
+
+    # Set environment variables so M1 mac
+    ENV['LIBRARY_PATH'] = "#{ENV['HOMEBREW_PREFIX']}/lib"
+    ENV['FFI_BUILD_FROM_SOURCE'] = '1'
+    cargo_bin = "#{IO.popen(['pwd']).read.strip}/extern/cargo_home/bin"
+    ENV['PATH'] += ":#{cargo_bin}"
+    puts "PATH = #{ENV['PATH']}"
+
+    # Build filecoin-ffi from source for Mac M1
+    system 'cd extern/filecoin-ffi && make'
+
+
+    system 'go mod download github.com/filecoin-project/go-state-types'
```

Comment:
Instead of needing this, shouldn't we be able to use another `replace` command for `go-state-types` ? See https://github.com/filecoin-project/lotus/blob/6f21a10a5601a8e2f24746fe7b75084c24b0864f/go.mod#L43

In go.mod of lotus install:  (could be added during homebrew install)
```golang
// in go.mod file:
replace github.com/filecoin-project/go-state-types => ./extern/go-state-types
```

```ruby
# in def install method
system "git clone --recurse-submodules https://github.com/filecoin-project/go-state-types.git extern/go-state-types"

```